src/commands/commerce/extension/map.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { flags, SfdxCommand } from '@salesforce/command';
+import { Messages, Org, SfdxError } from '@salesforce/core';
+import { forceDataSoql, forceDataRecordCreate } from '../../../lib/utils/sfdx/forceDataSoql';
+import { StatusFileManager } from '../../../lib/utils/statusFileManager';
+
+Messages.importMessagesDirectory(__dirname);
+
+const TOPIC = 'extension';
+const CMD = `commerce:${TOPIC}:map`;
+const msgs = Messages.loadMessages('@salesforce/commerce', 'store');
+
+export class MapExtension extends SfdxCommand {
+    public static readonly requiresUsername = true;
+
+    public static description = msgs.getMessage('extension.map.cmdDescription');
+    public static example = [`sfdx ${CMD} --registered-extension-name --store-id --store-name`];
+    public static flagsConfig = {
+        'registered-extension-name': flags.string({
+            char: 'r',
+            description: msgs.getMessage('extension.map.regExtensionNameFlagDescription'),
+        }),
+        'store-name': flags.string({
```

Comment:
What happens when the user does not provide both store name and Id? It should throw any error mentioning one of them is required